src/script/lua_api/l_object.cpp

```diff
@@ -170,16 +170,28 @@ int ObjectRef::l_punch(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
-	ObjectRef *puncher_ref = checkObject<ObjectRef>(L, 2);
+	ObjectRef *puncher_ref = nullptr;
+	try {
+		puncher_ref = checkObject<ObjectRef>(L, 2);
+	}
+	catch (...) {
+		puncher_ref = nullptr;
+	}
```

Comment:
it's bad practice to catch and discard exceptions like this
I suggest:
```
ObjectRef *puncher_ref = lua_isnoneornil(L, 2) ? nullptr : checkObject<ObjectRef>(L, 2);
```

Reply:
done